int64 right_tick
int64 left_tick
